### PR TITLE
Store configVersion in the job schema

### PR DESCRIPTION
**Pull request by despadam**
## Description

As seen in issue #1341

## Motivation

Since we are now using a global job configuration - instead of job specific - we should only store the configuration version inside each job in the db.

## Changes:

- `job.schema` was updated to only include `configVersion`
- In `jobs.controller`: `instanceAuthorizationJobCreate` and was updated to store `configVersion` in `jobInstance`

## Tests:

Tests should be updated to conform with the new job schema and retrieve the configuration from elsewhere.

### CODE REVIEW

File: src/jobs/jobs.controller.ts [hunk 1]
```diff
@@ -749,11 +737,11 @@ export class JobsController {
...
       request.user as JWTUser,
     );
-    const canCreate =
+    const canGet =
       ability.can(AuthOp.JobReadAny, JobClass) ||
       ability.can(AuthOp.JobReadAccess, currentJobInstance);
-    if (!canCreate) {
-      throw new ForbiddenException("Unauthorized to update this dataset");
+    if (!canGet) {
+      throw new ForbiddenException("Unauthorized to get this job.");
```
Comment (Junjiequan) [inline src/jobs/jobs.controller.ts]:
```
      const canView =
        ability.can(Action.DatasetReadAny, DatasetClass) ||
        ability.can(Action.DatasetReadOneOwner, datasetInstance) ||
        ability.can(Action.DatasetReadOneAccess, datasetInstance) ||
        ability.can(Action.DatasetReadOnePublic, datasetInstance);

      if (!canView) {
        throw new ForbiddenException("Unauthorized access");
      }
```
I suggest we either follow the variable namings for can ability from the `datasets.controller.ts`, or just to change all namings with canRead, canCreate. Having a vairious naming for the samething makes confusion sometimes.

**Review by Junjiequan — DISMISSED** [review]

File: src/jobs/jobs.controller.ts (hunk 1 above)
Comment (despadam) [reply src/jobs/jobs.controller.ts]: Covered in commit 4aedbc27d5667476a6d87cef1e06194d2be1b6df

File: src/jobs/jobs.controller.ts [hunk 2]
```diff
@@ -749,11 +737,11 @@ export class JobsController {
     const ability = this.caslAbilityFactory.createForUser(
       request.user as JWTUser,
     );
-    const canCreate =
+    const canRead =
```
Comment (bpedersen2) [inline src/jobs/jobs.controller.ts:748]: some test fail with permission problems, i suspect ti could be related to this change here

Comment (despadam) [reply src/jobs/jobs.controller.ts:748]: The latest commit fixes the failing tests

**Pull request merged**